core/include/kernel_defines.h

```diff
@@ -187,6 +187,20 @@ extern "C" {
  */
 #define IS_USED(module) IS_ACTIVE(module)
 
+/**
+ * @def         KERNEL_VERSION(mayor, minor, patch, extra)
+ * @brief       Generates a 64 bit variable of a release version
+ *              Comparisons to this only apply to released branches
+ *
+ * @param[in]   a   Mayor version of the release
+ * @param[in]   b   Minor version of the release
+ * @param[in]   c   Patch level of the release
+ * @param[in]   d   Extraversion, user defined
+ *
+ * @returns     A machine readable version variable
+ */
+#define KERNEL_VERSION(a, b, c, d) (((0ULL + a) << 48) + ((0ULL + b) << 32) + \
+                                    ((0ULL + c) << 16) + (d))
```

Comment:
```suggestion
#define KERNEL_VERSION(major, minor, patch, extra) \
    (((0ULL + major) << 48) + ((0ULL + minor) << 32) + \
     ((0ULL + patch) << 16) + (extra))
```

Reply:
Also still: we want to have a different name, I believe. `RIOT_VERSION_NUM` maybe?